test/e2e_node/services/namespace_controller.go

```diff
@@ -56,11 +59,15 @@ func (n *NamespaceController) Start() error {
 		return err
 	}
 	clientPool := dynamic.NewClientPool(config, registered.RESTMapper(), dynamic.LegacyAPIPathResolverFunc)
-	resources, err := client.Discovery().ServerPreferredNamespacedResources()
-	if err != nil {
-		return err
+	gvrFn := func() []k8s_unversioned.GroupVersionResource {
```

Comment:
I refactored this so it is a single line.  There really isn't a good library function to be made here, by moving the error handling down I get tests and a single line here...